isisd/isis_spf.c

```diff
@@ -515,14 +535,34 @@ isis_spf_add2tent(struct isis_spftree *spftree, enum vertextype vtype, void *id,
 	vertex->d_N = cost;
 	vertex->depth = depth;
 	if (VTYPE_IP(vtype) && psid) {
-		bool local;
+		struct isis_area *area = spftree->area;
+		struct isis_vertex lookup = {};
+		struct isis_vertex *vertex_psid;
 
-		local = (vertex->depth == 1);
-		vertex->N.ip.sr.sid = *psid;
-		vertex->N.ip.sr.label =
-			sr_prefix_in_label(spftree->area, psid, local);
-		if (vertex->N.ip.sr.label != MPLS_INVALID_LABEL)
-			vertex->N.ip.sr.present = true;
+		/*
+		 * Check if the Prefix-SID is already in use by another prefix.
+		 */
+		lookup.N.ip.sr.sid = *psid;
+		vertex_psid = hash_lookup(spftree->prefix_sids, &lookup);
+		if (vertex_psid
+		    && !prefix_same(&vertex_psid->N.ip.p.dest,
+				    &vertex->N.ip.p.dest)) {
+			zlog_warn(
+				"ISIS-Sr (%s): collision detected, Prefix-SID %u already in use by another prefix",
+				area->area_tag, psid->value);
```

Comment:
It will be also useful to print the prefix value to help operator debug the collision and the best it to print the Advertising router ID in addition to the prefix

Reply:
Good point. I've updated the error message to display more information about the Prefix-SID collision.